soh/src/code/z_parameter.c

```diff
@@ -2793,7 +2825,14 @@ void Inventory_UpdateBottleItem(PlayState* play, u8 item, u8 button) {
         item = ITEM_MILK_HALF;
     }
 
-    gSaveContext.inventory.items[gSaveContext.equips.cButtonSlots[button - 1]] = item;
+    if (CVarGetInteger("gRestoreRBAValues",0)) {
+        byteSwapInventory();
+        gSaveContext.inventory.items[gSaveContext.equips.cButtonSlots[button - 1]] = item;
+        byteSwapInventory();
```

Comment:
Correct me if I'm wrong, but is the need to byteswap before and after just to negate changes to the values not being updated by RBA?

I wonder if maybe instead we should explicitly only byteswap the value being updated by RBA.

Reply:
I was trying to think of ways where I would only swap the values being affected but to me it always ends up feeling bloated or gives off the same hard coded feeling I had before with just swapping every case individually. Regardless you would need to swap the value affected by RBA too as the game would otherwise interpret it incorrectly.